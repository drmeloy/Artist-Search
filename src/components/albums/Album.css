@import url('https://fonts.googleapis.com/css?family=Raleway:300&display=swap');
@import url('https://fonts.googleapis.com/css?family=Raleway:800&display=swap');
@import url('https://fonts.googleapis.com/css?family=Montserrat:300&display=swap');


.songName {
  text-decoration: none;
  color: white;
  font-family: 'Raleway' sans-serif;
  font-weight: 800;
  font-size: 30px;
}

