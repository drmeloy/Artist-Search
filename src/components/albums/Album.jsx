import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { useGetSongs } from '../hooks/songs';
import styles from './Album.css';

const Album = ({ match }) => {
  const { artistName, artistId, albumId } = match.params;

  const songs = useGetSongs(albumId); 

  const songList = songs.map(song => {
    return (
      <li key={song.id}>
        <Link  className={styles.songName} to={`/${artistName}/${artistId}/${albumId}/${song.title}`} >
          <p>{song.title}</p>
        </Link>
      </li>
    );});
  
  return (
    <ul>
      {songList}
    </ul>
  );
};

Album.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      artistName: PropTypes.string.isRequired,
      artistId: PropTypes.string.isRequired,
      albumId: PropTypes.string.isRequired
    }).isRequired
  }).isRequired
};

export default Album;
