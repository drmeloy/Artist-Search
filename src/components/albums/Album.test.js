import React from 'react';
import { shallow } from 'enzyme';
import Album from './Album';

describe('Album component', () => {
  it('matches its snapshot', () => {
    const wrapper = shallow(<Album match={{ params: { artistName: '', artistId: '', albumId: '' }}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
