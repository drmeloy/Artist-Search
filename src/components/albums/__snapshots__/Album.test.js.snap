// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album component matches its snapshot 1`] = `<ul />`;
